### PR TITLE
extrae: update checksums and fix the build (-lintl)

### PR DESCRIPTION
**Pull request by hainest**
It looks like the tarballs were updated after the checksums were added to spack.

### CODE REVIEW

**Review by bendamasio — CHANGES_REQUESTED** [review]
<details><summary>Hello @hainest! I encountered an install failure of extrae@4.0.6. Could you check it?</summary><br>
Hello @hainest,
I am terribly sorry, but my attempt on the installation of `extrae@4.0.6` failed in an automated build.
 But I've got a detailed report for you that may allow you to fix it based on the knowledge that you have about these recipes.

This command failed: `bin/spack install -v --fail-fast extrae@4.0.6`
<details><summary>Click here to drill down to the logs from building the failed spec: extrae@4.0.6</summary>
### `extrae@4.0.6`:
<details><summary><b>Click here to expand the full build log</b> for extrae@4.0.6</summary>
```py
==> extrae: Executing phase: 'autoreconf'
==> extrae: Executing phase: 'configure'
==> [2024-10-31-10:00:48.834947] Find (recursive): spack-build ['configure']
==> [2024-10-31-10:00:48.835317] Find complete: spack-build ['configure']
==> [2024-10-31-10:00:48.843540] '$PWD/.stage/spack-stage-extrae-4.0.6-wwbks7zxbw5bvnrrxu5o6tuyx2ouzkn2/spack-src/configure' 'LDFLAGS=-pthread' 'LIBS=-lintl' '--prefix=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/extrae-4.0.6-wwbks7zxbw5bvnrrxu5o6tuyx2ouzkn2' '--with-mpi=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openmpi-5.0.5-bges7exnb23r7z4c2kmearlbkxavbcnq' '--with-unwind=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/libunwind-1.6.2-kpyvmm3ioa6ombdujtaze3vxloxvpr2e' '--with-boost=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/boost-1.86.0-2ogleo2qf2qc7s3pjpuufw3irwguxyna' '--with-dwarf=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/libdwarf-0.11.0-c6zecyp22hv6ijjci4y6s22zsblfud5s' '--with-elf=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/elfutils-0.191-emogbtcvnz4cmd7sighurmxpiikjvl2i' '--with-xml-prefix=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libxml2-2.10.3-podbexn2oqikl336pjuib2i5k5hrwzsc' '--with-binutils=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/binutils-2.33.1-czg7d2kwwzxyetyc7uwopk4mromfwlwd' '--with-papi=$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/papi-7.1.0-x6fumkbiuq5tujrambkdkidnrdbk3fat' '--without-dyninst' '--without-cuda' '--without-cupti' '--disable-single-mpi-lib'
Configuring package Extrae 4.0.6
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking if this is an Altix machine... no
checking if this is a Cray XT machine... no
checking if this is a BG/L machine... no
checking if this is a BG/P machine... no
checking if this is a BG/Q machine... no
checking for /proc/self/maps... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '1000' is supported by ustar format... yes
checking whether GID '1000' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether $PWD/lib/spack/env/gcc/gcc accepts -g... yes
checking for $PWD/lib/spack/env/gcc/gcc option to enable C11 features... none needed
checking whether $PWD/lib/spack/env/gcc/gcc understands -c and -o together... yes
checking for C compiler vendor... gnu
checking for C compiler version... 13.1.0
checking whether the compiler supports GNU C++... yes
checking whether $PWD/lib/spack/env/gcc/g++ accepts -g... yes
checking for $PWD/lib/spack/env/gcc/g++ option to enable C++11 features... none needed
checking how to run the C preprocessor... $PWD/lib/spack/env/gcc/gcc -E
checking whether the compiler supports GNU Fortran... yes
checking whether $PWD/lib/spack/env/gcc/gfortran accepts -g... yes
checking how to get verbose linking output from $PWD/lib/spack/env/gcc/gfortran... -v
checking for Fortran libraries of $PWD/lib/spack/env/gcc/gfortran...  -L/usr/lib/gcc/x86_64-linux-gnu/13 -L/usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/13/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/13/../../.. -lintl -lgfortran -lm -lquadmath -lpthread -lintl:
checking for dummy main to link with Fortran libraries... unknown
configure: error: in `$PWD/.stage/spack-stage-extrae-4.0.6-wwbks7zxbw5bvnrrxu5o6tuyx2ouzkn2/spack-src/spack-build':
configure: error: linking to Fortran libraries from C fails
See `config.log' for more details
```
</details>

```py
checking for dummy main to link with Fortran libraries... unknown
configure: error: in `.stage/spack-stage-extrae-4.0.6-wwbks7zxbw5bvnrrxu5o6tuyx2ouzkn2/spack-src/spack-build':
configure: error: linking to Fortran libraries from C fails
See `config.log' for more details
==> Error: ProcessError: Command exited with status 1:
    .stage/spack-stage-extrae-4.0.6-wwbks7zxbw5bvnrrxu5o6tuyx2ouzkn2/spack-src/configure LDFLAGS=-pthread LIBS=-lintl --prefix=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/extrae-4.0.6-wwbks7zxbw5bvnrrxu5o6tuyx2ouzkn2 --with-mpi=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openmpi-5.0.5-bges7exnb23r7z4c2kmearlbkxavbcnq --with-unwind=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/libunwind-1.6.2-kpyvmm3ioa6ombdujtaze3vxloxvpr2e --with-boost=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/boost-1.86.0-2ogleo2qf2qc7s3pjpuufw3irwguxyna --with-dwarf=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/libdwarf-0.11.0-c6zecyp22hv6ijjci4y6s22zsblfud5s --with-elf=opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/elfutils-0.191-emogbtcvnz4cmd7sighurmxpiikjvl2i --with-xml-prefix=opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libxml2-2.10.3-podbexn2oqikl336pjuib2i5k5hrwzsc --with-binutils=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/binutils-2.33.1-czg7d2kwwzxyetyc7uwopk4mromfwlwd --with-papi=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/papi-7.1.0-x6fumkbiuq5tujrambkdkidnrdbk3fat --without-dyninst --without-cuda --without-cupti --disable-single-mpi-lib
     45    checking for dummy main to link with Fortran libraries... unknown
  >> 46    configure: error: in `$PWD/.stage/spack-stage-extrae-4.0.6-wwbks7zxbw5bvnrrxu5o6tuyx2ouzkn2/spack-src/spack-build':
  >> 47    configure: error: linking to Fortran libraries from C fails
     48    See `config.log' for more details
```
</details>

</details>

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
Excerpt of the config.log from @x64-Linux-Builder that causes a bunch of link tests to fail, including this final failure:
```
configure:9073: /home/bkaindl/spack.2/lib/spack/env/gcc/gcc -o conftest -g -O2  -pthread conftest.c -lintl  -L/usr/lib/gcc/x86_64-linux-gnu/13 -L/usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/13/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/13/../../.. -lintl -lgfortran -lm -lquadmath -lpthread -lintl: >&5
/usr/bin/ld: cannot find -lintl:: No such file or directory
collect2: error: ld returned 1 exit status
```
This happens when `libintl` which is provided by `gettext` is not installed on the host, and the configure script does not search `spack`'s library search path. In this case, we've to force the configure script to use `spack`'s library search path, at least for `gettext`.

`gettext` is a general pain point as Glibc integrated `libintl` so on Glibc systems you don't even need `-lintl` and also don't need any `-L`, as it's the system's Glibc that provides those `libintl` functions. But on non-Glibc systems such as Darwin, you need it. The fixup is a simple one-liner for the configure flags to add the library directory of the gettext built by (or imported externally imported into) spack. I push the commit that fixes the build for me (based on many other packages needing this fixup)

**Review by bendamasio — APPROVED** [review]
:information_source: `spack install` on the changed recipes of this PR was successful[^1] on Ubuntu 22.04.4 LTS!<details><summary>Click here for a summary of the successful builds.</summary>
These changed specs were found using [gh pr diff](https://cli.github.com):
- `extrae@4.0.6`
- `extrae@4.0.4`
- `extrae@4.0.3`
- `extrae@3.8.3`

<details><summary><b>Click here to show/hide the report on </b><tt>spack find</tt> for <tt>linux-ubuntu22.04-skylake / gcc@13.1.0</tt>.</summary><ol>
<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>extrae@3.8.3~cuda~cupti~dyninst+papi~single-mpi-lib build_system=autotools
>```
</summary>
>```py
>	autoconf@2.72
>		gmake@4.4.1
>		perl@5.40.0cpanmopcodeopenthreads
>			berkeley-db@18.1.40cxx,b231fcc
>			gdbm@1.23
>				readline@8.2
>	automake@1.16.5
>	binutils@2.33.1 compress_debug_sections=zlib
>		diffutils@3.10
>			libiconv@1.17
>		pkgconf@2.2.0
>		zlib-ng@2.2.1compatnew_strategiesopt
>	boost@1.86.0atomic_time_parallel_options_erasure cxxstd=11 visibility=hidden
>		bzip2@1.0.8
>		xz@5.4.6
>		zstd@1.5.6programs
>	elfutils@0.191
>	gcc-runtime@13.1.0
>	gettext@0.22.5bzip2cursesgit
>		ncurses@6.5
>		tar@1.34 zip=pigz
>			pigz@2.8
>	glibc@2.35
>	gmake@4.4.1
>	libdwarf@0.11.0decompressiondwarfdump
>		cmake@3.30.2
>			curl@8.8.0 tls=openssl
>				nghttp2@1.63.0
>	libtool@2.4.7
>		findutils@4.9.0
>		m4@1.4.19sigsegv,bfdffa7
>			libsigsegv@2.14
>	libunwind@1.6.2_signals_checks_exceptions_framedocs_backtrace
>	libxml2@2.10.3
>	m4@1.4.19sigsegv,bfdffa7
>		diffutils@3.10
>		libsigsegv@2.14
>	numactl@2.0.18
>	openmpi@5.0.5atomics-hwloc-libevent-pmix-rpath
>		hwloc@2.11.1-level-zero
>			libpciaccess@0.17
>				util-macros@1.20.1
>		libevent@2.1.12openssl
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>		openssh@9.8p1gssapi
>			krb5@1.21.2
>				bison@3.8.2
>				findutils@4.9.0
>			libedit@3.1-20240808
>			libxcrypt@4.4.35_api
>		pkgconf@2.2.0
>		pmix@5.0.3
>	papi@7.1.0_smi_tools
>
>```

</details>

<li><details><summary>Click on this area to unfold/refold the dependency report of this build:
>```py
>extrae@4.0.3~cuda~cupti~dyninst+papi~single-mpi-lib build_system=autotools
>```
</summary>
>```py
>	autoconf@2.72
>		gmake@4.4.1
>		perl@5.40.0cpanmopcodeopenthreads
>			berkeley-db@18.1.40cxx,b231fcc
>			gdbm@1.23
>				readline@8.2
>	automake@1.16.5
>	binutils@2.33.1 compress_debug_sections=zlib
>		diffutils@3.10
>			libiconv@1.17
>		pkgconf@2.2.0
>		zlib-ng@2.2.1compatnew_strategiesopt
>	boost@1.86.0atomic_time_parallel_options_erasure cxxstd=11 visibility=hidden
>		bzip2@1.0.8
>		xz@5.4.6
>		zstd@1.5.6programs
>	elfutils@0.191
>	gcc-runtime@13.1.0
>	gettext@0.22.5bzip2cursesgit
>		ncurses@6.5
>		tar@1.34 zip=pigz
>			pigz@2.8
>	glibc@2.35
>	gmake@4.4.1
>	libdwarf@0.11.0decompressiondwarfdump
>		cmake@3.30.2
>			curl@8.8.0 tls=openssl
>				nghttp2@1.63.0
>	libtool@2.4.7
>		findutils@4.9.0
>		m4@1.4.19sigsegv,bfdffa7
>			libsigsegv@2.14
>	libunwind@1.6.2_signals_checks_exceptions_framedocs_backtrace
>	libxml2@2.10.3
>	m4@1.4.19sigsegv,bfdffa7
>		diffutils@3.10
>		libsigsegv@2.14
>	numactl@2.0.18
>	openmpi@5.0.5atomics-hwloc-libevent-pmix-rpath
>		hwloc@2.11.1-level-zero
>			libpciaccess@0.17
>				util-macros@1.20.1
>		libevent@2.1.12openssl
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>		openssh@9.8p1gssapi
>			krb5@1.21.2
>				bison@3.8.2
>				findutils@4.9.0
>			libedit@3.1-20240808
>			libxcrypt@4.4.35_api
>		pkgconf@2.2.0
>		pmix@5.0.3
>	papi@7.1.0_smi_tools
>
>```
</details>

<li><details><summary>Click on this area to unfold/refold the dependency report of this build:
>```py
>extrae@4.0.4~cuda~cupti~dyninst+papi~single-mpi-lib build_system=autotools
>```
</summary>
>```py
>	autoconf@2.72
>		gmake@4.4.1
>		perl@5.40.0cpanmopcodeopenthreads
>			berkeley-db@18.1.40cxx,b231fcc
>			gdbm@1.23
>				readline@8.2
>	automake@1.16.5
>	binutils@2.33.1 compress_debug_sections=zlib
>		diffutils@3.10
>			libiconv@1.17
>		pkgconf@2.2.0
>		zlib-ng@2.2.1compatnew_strategiesopt
>	boost@1.86.0atomic_time_parallel_options_erasure cxxstd=11 visibility=hidden
>		bzip2@1.0.8
>		xz@5.4.6
>		zstd@1.5.6programs
>	elfutils@0.191
>	gcc-runtime@13.1.0
>	gettext@0.22.5bzip2cursesgit
>		ncurses@6.5
>		tar@1.34 zip=pigz
>			pigz@2.8
>	glibc@2.35
>	gmake@4.4.1
>	libdwarf@0.11.0decompressiondwarfdump
>		cmake@3.30.2
>			curl@8.8.0 tls=openssl
>				nghttp2@1.63.0
>	libtool@2.4.7
>		findutils@4.9.0
>		m4@1.4.19sigsegv,bfdffa7
>			libsigsegv@2.14
>	libunwind@1.6.2_signals_checks_exceptions_framedocs_backtrace
>	libxml2@2.10.3
>	m4@1.4.19sigsegv,bfdffa7
>		diffutils@3.10
>		libsigsegv@2.14
>	numactl@2.0.18
>	openmpi@5.0.5atomics-hwloc-libevent-pmix-rpath
>		hwloc@2.11.1-level-zero
>			libpciaccess@0.17
>				util-macros@1.20.1
>		libevent@2.1.12openssl
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>		openssh@9.8p1gssapi
>			krb5@1.21.2
>				bison@3.8.2
>				findutils@4.9.0
>			libedit@3.1-20240808
>			libxcrypt@4.4.35_api
>		pkgconf@2.2.0
>		pmix@5.0.3
>	papi@7.1.0_smi_tools
>
>```
</details>

<li><details><summary>Click on this area to unfold/refold the dependency report of this build:
>```py
>extrae@4.0.6~cuda~cupti~dyninst+papi~single-mpi-lib build_system=autotools
>```
</summary>
>```py
>	autoconf@2.72
>		gmake@4.4.1
>		perl@5.40.0cpanmopcodeopenthreads
>			berkeley-db@18.1.40cxx,b231fcc
>			gdbm@1.23
>				readline@8.2
>	automake@1.16.5
>	binutils@2.33.1 compress_debug_sections=zlib
>		diffutils@3.10
>			libiconv@1.17
>		pkgconf@2.2.0
>		zlib-ng@2.2.1compatnew_strategiesopt
>	boost@1.86.0atomic_time_parallel_options_erasure cxxstd=11 visibility=hidden
>		bzip2@1.0.8
>		xz@5.4.6
>		zstd@1.5.6programs
>	elfutils@0.191
>	gcc-runtime@13.1.0
>	gettext@0.22.5bzip2cursesgit
>		ncurses@6.5
>		tar@1.34 zip=pigz
>			pigz@2.8
>	glibc@2.35
>	gmake@4.4.1
>	libdwarf@0.11.0decompressiondwarfdump
>		cmake@3.30.2
>			curl@8.8.0 tls=openssl
>				nghttp2@1.63.0
>	libtool@2.4.7
>		findutils@4.9.0
>		m4@1.4.19sigsegv,bfdffa7
>			libsigsegv@2.14
>	libunwind@1.6.2_signals_checks_exceptions_framedocs_backtrace
>	libxml2@2.10.3
>	m4@1.4.19sigsegv,bfdffa7
>		diffutils@3.10
>		libsigsegv@2.14
>	numactl@2.0.18
>	openmpi@5.0.5atomics-hwloc-libevent-pmix-rpath
>		hwloc@2.11.1-level-zero
>			libpciaccess@0.17
>				util-macros@1.20.1
>		libevent@2.1.12openssl
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>		openssh@9.8p1gssapi
>			krb5@1.21.2
>				bison@3.8.2
>				findutils@4.9.0
>			libedit@3.1-20240808
>			libxcrypt@4.4.35_api
>		pkgconf@2.2.0
>		pmix@5.0.3
>	papi@7.1.0_smi_tools
>
>```
</details>

</details>

[^1]: Orchestrated using `spack install` by [github.com/spack/gh-spack-pr](https://github.com/spack/gh-spack-pr) and the [GitHub CLI](https://cli.github.com/)
</details>

**Review by bernhardkaindl — APPROVED** [review]

**Comment by hainest** [comment]
@bernhardkaindl I try to keep my pull requests as atomic as possible. I am more than happy to create additional pull requests when we encounter build errors. Always feel free to push that work down to me! :)

File: var/spack/repos/builtin/packages/extrae/package.py [hunk 1]
```diff
@@ -116,34 +116,37 @@ def configure_args(self):
...
             ["--with-dyninst=%s" % spec["dyninst"].prefix]
-            if "+dyninst" in self.spec
+            if spec.satisfies("+dyninst")
             else ["--without-dyninst"]
         )
 
         args += (
             ["--with-cuda=%s" % spec["cuda"].prefix]
-            if "+cuda" in self.spec
+            if spec.satisifes("+cuda")
```
Comment (giordano) [inline var/spack/repos/builtin/packages/extrae/package.py:131]: `satisifes`?

**Comment by haampie** [comment]
@bernhardkaindl run your bot on your own commits too ;)

**Pull request merged**